x-pack/legacy/plugins/reporting/server/lib/get_user.ts

```diff
@@ -4,16 +4,11 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { Legacy } from 'kibana';
 import { KibanaRequest } from '../../../../../../src/core/server';
 import { ReportingSetupDeps } from '../types';
-import { LevelLogger } from './level_logger';
 
-export function getUserFactory(security: ReportingSetupDeps['security'], logger: LevelLogger) {
-  /*
-   * Legacy.Request because this is called from routing middleware
-   */
-  return async (request: Legacy.Request) => {
-    return security?.authc.getCurrentUser(KibanaRequest.from(request)) ?? null;
+export function getUserFactory(security: ReportingSetupDeps['security']) {
```

Comment:
NIT: I would directly use `security?: SecurityPluginSetup` instead of `ReportingSetupDeps['security']` here.